tests/results/ResultFilterTest.py

```diff
@@ -562,3 +562,87 @@ def test_delete_file_with_result(self):
                                      [old_result_tf1, old_result_tf2],
                                      [new_result])
         self.assertEqual(new_results, [new_result])
+
```

Comment:
can you maybe put all tests that test the `ensure_files_present` in an own class? Then we can rely on them not depending on anything that's needed for the others, less complexity to think about.